docs/yaml/functions/subproject.yaml

```
@@ -8,12 +8,12 @@ description: |
   example a subproject called `foo` must be located in
   `${MESON_SOURCE_ROOT}/subprojects/foo`.
 
-  - `default_options` *(since 0.37.0)*: an array of default option values
-    that override those set in the subproject's `meson_options.txt`
-    (like `default_options` in `project`, they only have effect when
-    Meson is run for the first time, and command line arguments override
-    any default options in build files). *(since 0.54.0)*: `default_library`
-    built-in option can also be overridden.
+  - `default_options` *(since 0.37.0)*: an array or dict of default option
```

```suggestion
  - `default_options` *(since 0.37.0)*: an array of default option
```
I think this would be confusing, as the dict is allowed only since 1.1.0.